contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+contract DomainRegistry {
+    uint256 constant REQUIRED_DEPOSIT = 1 ether; 
+    
+    
+
+    struct Domain {
+        address controller; 
+        string tld; 
+        uint256 createdTimestamp; 
+        uint256 deposit; 
+        bool isRegistered; 
+        
+    }
+
+
+    // Мапа для зберігання доменів, де ключ - це назва домену верхнього рівня
+    mapping(string => Domain) public domains;
+
+    // Подія, яка виникає при створенні нового домену
+    event DomainCreated(address indexed controller, string indexed tld, uint256 createdTimestamp, uint256 deposit);
+
+    // Модифікатор перевірки застави
+    modifier hasRequiredDeposit(uint256 _requiredDeposit) {
+        require(msg.value >= _requiredDeposit, "Wrong eth amount");
+         _;
+    }
+
+    modifier domainDoesNotExist(string memory _tld) {
+        require(!domains[_tld].isRegistered, "Domain exists");
+        _;
+    }
+
+modifier isValidDomain(string memory _tld) {
+    require(keccak256(bytes(_tld)) == keccak256(bytes("com")) || keccak256(bytes(_tld)) == keccak256(bytes("gov")), "Wrong domain level");
+    _;
+}
```

Comment:
What is the purpose for this validator logic? -5

Reply:
- add validation for top level domains
- add tests